### PR TITLE
Add Kakao OAuth2 client

### PR DESCRIPTION
**Pull request by joonas-yoon**
I made some client for Kakao web service.

it served by Kakao corporation and they have many services on it. I think [this document](https://www.kakaocorp.com/page/?lang=ENG) and [this one on wikipedia](https://en.wikipedia.org/wiki/Kakao) is helpful.

also I wrote pytest codes and all passed.

FYI, I am not a member of that company and have no relation with the company.

### CODE REVIEW

File: httpx_oauth/clients/kakao.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                raise GetIdEmailError(response.json())
+
+            account_info = cast(Dict[str, Any], response.json())
+            account_id = str(account_info.get('id'))
+            kakao_account = account_info.get('kakao_account')
+
+            try:
+                return account_id, kakao_account.get('email')
+            except:
+                return account_id, None
```
Comment (frankie567) [inline httpx_oauth/clients/kakao.py:63]: This line misses coverage, but actually, it's not necessary.

When you call `.get` on a dictionary, it's already a safe method: if the key doesn't exist, it'll return `None`.

So you can directly do:

```py
return account_id, kakao_account.get('email')
```

without a `try`/`except`.

**Comment by frankie567** [comment]
HI @joonas-yoon 👋 

Looks good! There is a small tweak to make and we should be good to go!

File: httpx_oauth/clients/kakao.py (hunk 1 above)
Comment (joonas-yoon) [reply httpx_oauth/clients/kakao.py:63]: woah i didn't know that, thanks for your information!

I will fix and make a commit.

**Comment by frankie567** [comment]
@all-contributors add @joonas-yoon for code

**Pull request merged**